### PR TITLE
feat: ensure tags have the same starting and ending names

### LINKED ISSUE #8: feature: match start and end tags
### Did you check the tree-sitter docs?

- [X] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser

### Is your feature request related to a problem? Please describe.

`<From>Jani</from>` is invalid in XML but the parser allows it.

### Describe the solution you'd like

The parser should ensure that start tags and end tags have the same name.

### Describe alternatives you've considered

Just leave it to the language server/linter.

### Additional context

Specification: https://www.w3.org/TR/xml/#sec-starttags
Example: https://www.w3schools.com/xml/note_error.xml

### PR DESCRIPTION
**Pull request by amaanq**
Closes #8

### CODE REVIEW

File: xml/src/scanner.c [hunk 1]
```diff
@@ -1,18 +1,271 @@
...
+                VEC_PUSH(scanner->tags, tag);
+            }
+        }
+    }
+}
+
+static String scan_tag_name(TSLexer *lexer) {
+    String tag_name = string_new();
+    while (iswalnum(lexer->lookahead) || lexer->lookahead == '-' || lexer->lookahead == ':') {
+        STRING_PUSH(tag_name, towupper(lexer->lookahead));
```
Comment (ObserverOfTime) [inline xml/src/scanner.c:166]: XML tags are case-sensitive. I'll fix this along with some nitpicks.

**Comment by amaanq** [comment]
@ObserverOfTime macos is annoying in CI, can we just use iswalpha since wasm allows for the wide character versions of these symbols by default?

**Comment by ObserverOfTime** [comment]
Does XML support the wide character versions?

**Comment by amaanq** [comment]
idk, but we can do `lexer->lookahead < UINT8_MAX && iswalpha(lexer->lookahead)`

File: xml/grammar.js [hunk 2]
```diff
@@ -117,13 +122,20 @@ module.exports = grammar(DTD, {
...
+      alias($._start_tag_name, $.Name),
       c.rseq($._S, $.Attribute),
       O($._S),
       '>'
     ),
 
-    ETag: $ => seq('</', $.Name, O($._S), '>'),
+    ETag: $ => seq('</', alias($._end_tag_name, $.Name), O($._S), '>'),
+
+    _ErroneousETag: $ => seq(
```
Comment (ObserverOfTime) [inline xml/grammar.js:133]: Is it possible to replace this with an `ERROR` node?

**Comment by ObserverOfTime** [comment]
I'll figure out how to fix the examples action…

File: xml/grammar.js (hunk 2 above)
Comment (amaanq) [reply xml/grammar.js:133]: not exactly, you could alias it though

Comment (ObserverOfTime) [reply xml/grammar.js:133]: Would tree-sitter count it as an error node?

Comment (amaanq) [reply xml/grammar.js:133]: nah it wouldn't, if you mean the error symbol under the hood

**Pull request merged**